src/libp2p/service.rs

```diff
@@ -484,6 +485,12 @@ async fn handle_network_message(
                         warn!("Failed to get Libp2p listeners");
                     }
                 }
+                NetRPCMethods::Peer(response_channel, peer) => {
+                    let addresses = swarm.behaviour().peer_addresses().get(&peer).cloned();
+                    if response_channel.send(addresses).is_err() {
+                        warn!(%peer, "Failed to get Libp2p peer addresses");
```

Comment:
Is it a failure in obtaining the p2p address or is this a channel fiasco, e.g., dropped?

Reply:
It's a channel fiasco. The oneshot API does not return an error tho
```rust
pub fn send(self, t: T) -> Result<(), T>
```

Reply:
the error message should be adjusted, then, correct?

Reply:
Fixed for all the variants by switching to `flume`